src/main/java/fi/aalto/cs/apluscourses/ui/AboutDialog.java

```diff
@@ -24,27 +29,47 @@ private AboutDialog() {
   /**
    * Displays a small pop up window containing basic information about the plugin.
    */
-  public static void display() {
+  public static void display() throws IOException {
+
+    final Icon icon = PluginIcons.A_PLUS_LOGO;
     JOptionPane.showMessageDialog(null,
         createAboutInnerContainer(),
         getText("ui.aboutDialog.title"),
-        JOptionPane.INFORMATION_MESSAGE);
+        JOptionPane.INFORMATION_MESSAGE, icon);
   }
 
   @NotNull
-  private static JPanel createAboutInnerContainer() {
+  private static JPanel createAboutInnerContainer() throws IOException {
     JPanel panel = new JPanel();
     panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
```

Comment:
I like this way of building the view. We have elsewhere used mainly IntelliJ's visual GUI designer but it's a bit clumsy and XML it produces is messy. This is more structured and VCS friendly way.

I see no remarkable problem in having different view-building approaches in our code base; they're after all just implementation details of each view class. :+1: